File: Examples/VisualBasic/Exchange/PagingSupportForListingFolders.vb
Imports Aspose.Email.Exchange

Namespace Aspose.Email.Examples.VisualBasic.Email.Exchange

    Public Class PagingSupportForListingFolders
        Private Shared Sub Run()
            'ExStart: PagingSupportForListingFolders
            '<summary>
            'This example shows how to retrieve folders information from Exchange Server with Paging Support
            'Introduced in Aspose.Email for .NET 6.4.0
            '</summary>
            Using client As IEWSClient = EWSClient.GetEWSClient("exchange.domain.com", "username", "password")
                Dim itemsPerPage As Integer = 5
                Dim totalFoldersCollection As ExchangeFolderInfoCollection = client.ListSubFolders(client.MailboxInfo.RootUri)
                Console.WriteLine(totalFoldersCollection.Count)

                '/ RETREIVING INFORMATION USING PAGING SUPPORT 

                Dim pages As New List(Of ExchangeFolderPageInfo)()
                Dim pagedFoldersCollection As ExchangeFolderPageInfo = client.ListSubFoldersByPage(client.MailboxInfo.RootUri, itemsPerPage)

                Console.WriteLine(pagedFoldersCollection.TotalCount)

                pages.Add(pagedFoldersCollection)
                While Not pagedFoldersCollection.LastPage
                    pagedFoldersCollection = client.ListSubFoldersByPage(client.MailboxInfo.RootUri, itemsPerPage, pagedFoldersCollection.PageOffset + 1)
                    pages.Add(pagedFoldersCollection)
                End While
                Dim retrievedFolders As Integer = 0
                For Each pageCol As ExchangeFolderPageInfo In pages
                    retrievedFolders += pageCol.Items.Count
                Next

                'Verify the total count of folders
                Console.WriteLine(retrievedFolders)
            End Using
            'ExEnd: PagingSupportForListingFolders
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Exchange_EWS/PagingSupportForListingFolders.cs
using System;
using System.Collections.Generic;
using Aspose.Email.Clients.Exchange.WebService;
using Aspose.Email.Clients.Exchange;

namespace Aspose.Email.Examples.CSharp.Exchange
{
    class PagingSupportForListingFolders
    {
        static void Run()
        {
            // ExStart: PagingSupportForListingFolders
            ///<summary>
            /// This example shows how to retrieve folders information from Exchange Server with Paging Support
            /// Introduced in Aspose.Email for .NET 6.4.0
            ///</summary>
            using (IEWSClient client = EWSClient.GetEWSClient("exchange.domain.com", "username", "password"))
            {
                int itemsPerPage = 5;
                ExchangeFolderInfoCollection totalFoldersCollection = client.ListSubFolders(client.MailboxInfo.RootUri);
                Console.WriteLine(totalFoldersCollection.Count);

                //////////////////// RETREIVING INFORMATION USING PAGING SUPPORT //////////////////////////////////

                List<ExchangeFolderPageInfo> pages = new List<ExchangeFolderPageInfo>();
                ExchangeFolderPageInfo pagedFoldersCollection = client.ListSubFoldersByPage(client.MailboxInfo.RootUri, itemsPerPage);

                Console.WriteLine(pagedFoldersCollection.TotalCount);

                pages.Add(pagedFoldersCollection);
                while (!pagedFoldersCollection.LastPage)
                {
                    pagedFoldersCollection = client.ListSubFoldersByPage(client.MailboxInfo.RootUri, itemsPerPage, pagedFoldersCollection.PageOffset + 1);
                    pages.Add(pagedFoldersCollection);
                }
                int retrievedFolders = 0;
                foreach (ExchangeFolderPageInfo pageCol in pages)
                    retrievedFolders += pageCol.Items.Count;

                // Verify the total count of folders
                Console.WriteLine(retrievedFolders);
            }            
            // ExEnd: PagingSupportForListingFolders
        }
    }
}
